pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/ThreadRuntimeFactory.java

```diff
@@ -44,15 +53,25 @@
     private final FunctionCacheManager fnCache;
     private final PulsarClient pulsarClient;
     private final String storageServiceUrl;
+    private final SecretsProvider secretsProvider;
     private volatile boolean closed;
 
     public ThreadRuntimeFactory(String threadGroupName, String pulsarServiceUrl, String storageServiceUrl,
-            AuthenticationConfig authConfig) throws Exception {
-        this(threadGroupName, createPulsarClient(pulsarServiceUrl, authConfig), storageServiceUrl);
+                                AuthenticationConfig authConfig, String secretsProviderClassName,
+                                Map<String, String> secretsProviderConfig) throws Exception {
+        this(threadGroupName, createPulsarClient(pulsarServiceUrl, authConfig), storageServiceUrl, secretsProviderClassName, secretsProviderConfig);
     }
 
     @VisibleForTesting
-    public ThreadRuntimeFactory(String threadGroupName, PulsarClient pulsarClient, String storageServiceUrl) {
+    public ThreadRuntimeFactory(String threadGroupName, PulsarClient pulsarClient, String storageServiceUrl,
+                                String secretsProviderClassName, Map<String, String> secretsProviderConfig) {
+        try {
+            secretsProvider = (SecretsProvider) Reflections.createInstance(secretsProviderClassName, ClassLoader.getSystemClassLoader());
```

Comment:
I don't think this should be enforced it's an optional thing and no implementation should be the default. Also just enable this for kubernetes runtime to start then add the others in additional patches.

Reply:
In general it's not a good idea to throw exceptions in a constructor. Is only the secretsProvider object needed in the class? Can the instantiation and configuration happen outside in utility or factory method and then passed in?

Reply:
Done